src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -123,36 +126,37 @@ public Optional<EstimatedRobotPose> getEstimatedGlobalPose(Pose2d prevEstimatedR
         return m_photonPoseEstimator.update();
     }
 
-	/*public Pose2d getTagPose() {
-		var targetResult = m_aprilTagCamera.getLatestResult();
-		
-		int targetID = -1;
-		if (targetResult.hasTargets()) {
-			// Get the current best target.
-			PhotonTrackedTarget target = targetResult.getBestTarget();
-			targetID = target.getFiducialId();
-		}
-		return m_aprilTagFieldLayout.getTagPose(targetID).get().toPose2d();
-
-	}*/
-    //returns ID of best target
-    public int getBestTagID(){
+    public Pose2d getCentralTagPose() { // gets target closets to center of camera
         var targetResult = m_aprilTagCamera.getLatestResult();
-		
-		int targetID = -1;
-		if (targetResult.hasTargets()) {
-			// Get the current best target.
-			PhotonTrackedTarget target = targetResult.getBestTarget();
-			targetID = target.getFiducialId();
-		}
-        return targetID; 
+        int targetID = -1;
+        if (targetResult.hasTargets()) {
+            // Get lists of targets
+            List<PhotonTrackedTarget> targets = targetResult.getTargets();
+            // make a temp holder var for least Y translation
+            Translation2d translationHolder = new Translation2d(0, 0);
+            for (PhotonTrackedTarget tag : targets) { // for every target in camera
+                // get transformation to target
+                Transform3d tagTransform = tag.getBestCameraToTarget();
+                // get translation to target from transformation
+                Translation2d tagTranslation = tagTransform.getTranslation().toTranslation2d();
+
+                // looking for smallest absolute relative to camera Y
+                // if abs Y translation of new tag is less then holder tag, it becomes holder tag
+                if (Math.abs(tagTranslation.getY()) < Math.abs(translationHolder.getY())) {
```

Comment:
This is problematic with the line above: "Translation2d translationHolder = new Translation2d(0, 0);".

When you try to getY() for this translationHolder, you always get 0, and it is impossible for any later translations to have an absolute Y value less than 0, which makes the for loop doesn't do anything because the translationHolder never gets updated. 

The fix would be probably to simply assign the first tag position to the translationHolder and don't compare for the first loop.